frontend/src/modules/notification/reducers.ts

```diff
@@ -0,0 +1,19 @@
+import { Reducer } from "redux"
+import { ALERT_REMOVE, AlertAction, AlertRemoveAction, ALERT, NotificationState } from "./models"
+import {del, merge} from 'object-path-immutable'
+
+export const notifications: Reducer<NotificationState, AlertAction | AlertRemoveAction> = (state, action) => {
+    switch(action.type) {
+        case ALERT: {
+            const alertAction = action as AlertAction
+            return merge<NotificationState>(state ?? {}, [alertAction.id], alertAction.props)
```

Comment:
It's important that alerts are displayed in the correct and consistent order (newest to oldest). It might be better to use an array to store them in order.

If a newer notification replaces an existing one with the same ID it should appear as the most recent alert and the older one should disappear. If we just use object properties then the it will not be promoted to the most recent alert, as the order of the keys won't change.